### PR TITLE
Move all CUDA code from plugins to libraries

### PR DESCRIPTION
**Pull request by fwyzard**

### CODE REVIEW

**Comment by fwyzard** [comment]
As far as I can tell, this should remove all references to CUDA device code and kernels from all the plugins - but it does not fix the `invalid device function` error.

File: RecoPixelVertexing/PixelTriplets/interface/GPUCACell.h [hunk 1]
```diff
@@ -4,8 +4,8 @@
 #ifndef GPU_CACELL_H_
 #define GPU_CACELL_H_
 
-#include "GPUHitsAndDoublets.h"
-#include "RecoLocalTracker/SiPixelRecHits/plugins/siPixelRecHitsHeterogeneousProduct.h"
+#include "RecoPixelVertexing/PixelTriplets/interface/GPUHitsAndDoublets.h"
```
Comment (makortel) [inline RecoPixelVertexing/PixelTriplets/interface/GPUCACell.h]: To me it looks like this include is not needed. Actually nothing from `GPUHitsAndDoublets.h` appears to be used at the moment?

File: RecoPixelVertexing/PixelTriplets/interface/gpuPixelDoublets.h [hunk 2]
```diff
@@ -9,9 +9,9 @@
 #include <limits>
 
 #include "DataFormats/Math/interface/approx_atan2.h"
-#include "RecoLocalTracker/SiPixelRecHits/plugins/siPixelRecHitsHeterogeneousProduct.h"
+#include "RecoLocalTracker/SiPixelRecHits/interface/siPixelRecHitsHeterogeneousProduct.h"
 
-#include "GPUCACell.h"
+#include "RecoPixelVertexing/PixelTriplets/interface/GPUCACell.h"
```
Comment (makortel) [inline RecoPixelVertexing/PixelTriplets/interface/gpuPixelDoublets.h]: Also here forward-declaring `GPUCACell` would be sufficient.

`gpuPixelDoublets.h` itself seems not be used outside `RecoPixelVertexing/PixelTriplets/src`, so I'd move it to the `src`.

File: HeterogeneousCore/CUDAServices/plugins/BuildFile.xml [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
-<use   name="FWCore/PluginManager"/>
-<use   name="FWCore/ServiceRegistry"/>
-<use   name="FWCore/Utilities"/>
-<use   name="FWCore/Concurrency"/>
-<use   name="FWCore/Framework"/>
-<use   name="HeterogeneousCore/CUDAServices"/>
-<use   name="cuda"/>
-<library   file="*.cc" name="HeterogeneousCoreCUDAServicesPlugins">
-  <flags   EDM_PLUGIN="1"/>
+<use name="cuda"/>
```
Comment (makortel) [inline HeterogeneousCore/CUDAServices/plugins/BuildFile.xml]: I suppose this is needed for `NVProfilerService`? Should it be moved to `src` as well?

Comment (fwyzard) [reply HeterogeneousCore/CUDAServices/plugins/BuildFile.xml]: Even if we don't `<use name="cuda"/>` in the plugins, the plugin library will depend on the main library, which depends on `cuda` - so we wouldn't gain anything AFAICT.

Comment (makortel) [reply HeterogeneousCore/CUDAServices/plugins/BuildFile.xml]: But what about the `NVProfilerService` itself?

**Comment by fwyzard** [comment]
This turned out not to work, and actually make the `invalid device function` error more frequent...

#123 goes in the opposite direction,and seems to fix the problem for the moment.

**Pull request closed without merging**